test/e2e/aks_fleets_member.go

```diff
@@ -0,0 +1,147 @@
+			"jobName":           ptr.To(os.Getenv(JobName)),
+			"creationTimestamp": ptr.To(os.Getenv(Timestamp)),
+		},
+	}, nil)
+	Expect(err).To(BeNil())
+
+	fleetClient, err := armcontainerservicefleet.NewFleetsClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	fleetsMemberClient, err := armcontainerservicefleet.NewFleetMembersClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("creating a fleet manager")
+	poller, err := fleetClient.BeginCreateOrUpdate(ctx, groupName, fleetName, armcontainerservicefleet.Fleet{
+		Location: ptr.To(os.Getenv(AzureLocation)),
+	}, nil)
+	Expect(err).To(BeNil())
+	_, err = poller.PollUntilDone(ctx, nil)
+	Expect(err).To(BeNil())
+
+	By("Joining the cluster to the fleet hub")
+	var infraControlPlane = &infrav1.AzureManagedControlPlane{}
+	Eventually(func(g Gomega) {
+		err = mgmtClient.Get(ctx, client.ObjectKey{Namespace: input.Cluster.Spec.ControlPlaneRef.Namespace, Name: input.Cluster.Spec.ControlPlaneRef.Name}, infraControlPlane)
+		g.Expect(err).NotTo(HaveOccurred())
+		infraControlPlane.Spec.FleetsMember = &infrav1.FleetsMember{
+			FleetsMemberClassSpec: infrav1.FleetsMemberClassSpec{
+				ManagerName:          fleetName,
+				ManagerResourceGroup: groupName,
+				Group:                updateGroupName,
+			},
+		}
+		g.Expect(mgmtClient.Update(ctx, infraControlPlane)).To(Succeed())
+	}, input.WaitIntervals...).Should(Succeed())
+
+	By("Ensuring the fleet member is created and attached to the managed cluster")
+	Eventually(func(g Gomega) {
+		resp, err := fleetsMemberClient.Get(ctx, groupName, fleetName, input.Cluster.Name, nil)
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(resp.Properties.ProvisioningState).To(Equal(ptr.To(armcontainerservicefleet.FleetMemberProvisioningStateSucceeded)))
+		fleetsMember := resp.FleetMember
+		g.Expect(fleetsMember.Properties).NotTo(BeNil())
+		expectedID := azure.ManagedClusterID(getSubscriptionID(Default), infraControlPlane.Spec.ResourceGroupName, input.Cluster.Name)
+		g.Expect(fleetsMember.Properties.ClusterResourceID).To(Equal(ptr.To(expectedID)))
+	}, input.WaitIntervals...).Should(Succeed())
+
+	By("Remove the FleetsMember spec from the AzureManagedControlPlane")
+	Eventually(func(g Gomega) {
+		err = mgmtClient.Get(ctx, client.ObjectKey{Namespace: input.Cluster.Spec.ControlPlaneRef.Namespace, Name: input.Cluster.Spec.ControlPlaneRef.Name}, infraControlPlane)
+		g.Expect(err).NotTo(HaveOccurred())
+		infraControlPlane.Spec.FleetsMember = nil
+		g.Expect(mgmtClient.Update(ctx, infraControlPlane)).To(Succeed())
+	}, input.WaitIntervals...).Should(Succeed())
+
+	By("Waiting for the FleetsMember to finish updating")
+	Eventually(func(g Gomega) {
+		resp, err := fleetsMemberClient.Get(ctx, groupName, fleetName, input.Cluster.Name, nil)
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(resp.Properties.ProvisioningState).NotTo(Equal(ptr.To(armcontainerservicefleet.FleetMemberProvisioningStateUpdating)))
```

Comment:
What does the provisioning state converge here to, out of curiosity?

And this may also be a case where https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/4338 would be beneficial if CAPZ currently provides no way to natively delete the fleetsmember without deleting the whole cluster.

Reply:
I'm not entirely sure, I just made this change to try and fix an error saying that the `fleetsmember` is stuck in the updating state when I try and delete it. I'm not sure why it is in that state when I remove it from the CAPZ spec. I think #4338 would definitely be beneficial in this case, otherwise it would be hard to remove the `fleetsmembers` from the fleet manager (you can't use CAPZ to do it and would have to do it manually through az cli or portal)

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/4316/pull-cluster-api-provider-azure-e2e-aks/1735809843710660608

Reply:
Oh I misread the error. It seems like the managed cluster itself is in an updating state. I should be checking for that instead.

Reply:
Or maybe taking a step back, what exactly do we expect to happen when deleting the fleet spec from an existing cluster? I'm actually not sure it going into an Updating state at all is very desirable if we don't know exactly what's changing. Overall, If that's not well-defined currently then I could be convinced to drop this from the test. And maybe add webhook validation that the fleet member is never deleted once it's created.

Reply:
I'm not sure why it would still be updating if the update on this line succeeded though: https://github.com/willie-yao/cluster-api-provider-azure/blob/aks-fleets-member/test/e2e/aks_fleets_member.go#L124

Reply:
that is a UX decision from cluster-api point of view IMO.
if you remove the member spec, do you want to leave the fleet?
then deleting the azure member resource is the right move.

note that a cluster can only be member of 1 fleet.
so if you remove the member spec, or change it to point at another fleet, that won't work, unless you delete the member, and create it again with the new values.

Reply:
>  Overall, If that's not well-defined currently then I could be convinced to drop this from the test. And maybe add webhook validation that the fleet member is never deleted once it's created.

I think we should delete the member since it is tied to the cluster. Also, leaving the fleetsmembers prevent the fleet manager from being deleted, although I think we can manually detach the members from the manager using the client.